theatre/studio/src/panels/SequenceEditorPanel/RightOverlay/HorizontalScrollbar.tsx

```diff
@@ -174,117 +175,143 @@ const HorizontalScrollbar: React.FC<{
+              start:
+                valuesBeforeDrag.clippedSpaceRange.start + deltaPosInUnitSpace,
+              end: valuesBeforeDrag.clippedSpaceRange.end,
+            }
+
+            if (newRange.start > newRange.end - 1) {
+              newRange.start = newRange.end - 1
+            }
+
+            if (newRange.start <= 0) {
+              newRange.start = 0
+            }
+
+            val(layoutP.clippedSpace.setRange)(newRange)
+          },
+          onDragEnd() {
+            setBeingDragged('nothing')
+          },
         }
-
-        val(layoutP.clippedSpace.setRange)(newRange)
       },
 
-      dragRangeEnd: (dx: number) => {
-        setBeingDragged('end')
+      onRangeEndDragStart() {
+        noteValuesBeforeDrag()
+        return {
+          onDrag(dx: number) {
+            setBeingDragged('end')
 
-        const deltaPosInUnitSpace = deltaXToDeltaPos(dx)
+            const deltaPosInUnitSpace = deltaXToDeltaPos(dx)
 
-        const newRange: IRange = {
-          start: valuesBeforeDrag.clippedSpaceRange.start,
-          end: valuesBeforeDrag.clippedSpaceRange.end + deltaPosInUnitSpace,
-        }
+            const newRange: IRange = {
+              start: valuesBeforeDrag.clippedSpaceRange.start,
+              end: valuesBeforeDrag.clippedSpaceRange.end + deltaPosInUnitSpace,
+            }
 
-        if (newRange.end < newRange.start + 1) {
-          newRange.end = newRange.start + 1
-        }
+            if (newRange.end < newRange.start + 1) {
+              newRange.end = newRange.start + 1
+            }
 
-        if (newRange.end >= valuesBeforeDrag.assumedLengthOfSequence) {
-          newRange.end = valuesBeforeDrag.assumedLengthOfSequence
-        }
+            if (newRange.end >= valuesBeforeDrag.assumedLengthOfSequence) {
+              newRange.end = valuesBeforeDrag.assumedLengthOfSequence
+            }
 
-        val(layoutP.clippedSpace.setRange)(newRange)
+            val(layoutP.clippedSpace.setRange)(newRange)
+          },
+          onDragEnd() {
+            setBeingDragged('nothing')
+          },
+        }
       },
     }
 
     return self
   }, [layoutP, relevantValuesD])
 
+  const [refRangeDrag, nodeRangeDrag] = useRefAndState<HTMLDivElement | null>(
+    null,
+  )
+  useDrag(nodeRangeDrag, {
+    debugName: 'HorizontalScrollbar/onRangeDrag',
+    onDragStart: handles.onRangeDragStart,
+    lockCSSCursorTo: 'ew-resize',
+  })
+
+  const [refRangeStartDrag, nodeRangeStartDrag] =
+    useRefAndState<HTMLDivElement | null>(null)
+  useDrag(nodeRangeStartDrag, {
+    debugName: 'HorizontalScrollbar/onRangeStartDrag',
+    onDragStart: handles.onRangeStartDragStart,
+    lockCSSCursorTo: 'w-resize',
+  })
+
+  const [refRangeEndDrag, nodeRangeEndDrag] =
+    useRefAndState<HTMLDivElement | null>(null)
+  useDrag(nodeRangeEndDrag, {
+    debugName: 'HorizontalScrollbar/onRangeEndDrag',
+    onDragStart: handles.onRangeEndDragStart,
+    lockCSSCursorTo: 'e-resize',
+  })
```

Comment:
@AndrewPrifer, is this where we can just `useState`?

Reply:
👍 Removed usages of useRefAndState in favour of useState.